src/api/app/controllers/webui/request_controller.rb

```diff
@@ -515,53 +540,39 @@ def staging_status(request, target_project)
     }
   end
 
-  def prepare_request_data
-    @is_author = @bs_request.creator == User.possibly_nobody.login
-    @is_target_maintainer = @bs_request.is_target_maintainer?(User.session)
-    reviews = @bs_request.reviews.where(state: 'new')
-    @my_open_reviews = reviews.select { |review| review.matches_user?(User.session) }
-    @can_add_reviews = @bs_request.state.in?([:new, :review]) && (@is_author || @is_target_maintainer || @my_open_reviews.present?)
+  def should_refresh(action)
```

Comment:
@ncounter hmm, maybe this is only required for the `changes` view/tab. I think this job is responsible for fetching and caching the source diff. Have you checked if this is really needed for the views other then `changes`?

Reply:
You're right!

On the other hand, now that are we thinking about it, I wonder if it would be nice to trigger it from the `header_data`, just to prepare earlier the diff data whenever someone enters for the first time the request page. But that's something we can adjust later, I'll start dropping it from `build_results` and `rpm_lint`.